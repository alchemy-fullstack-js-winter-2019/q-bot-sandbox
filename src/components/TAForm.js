import React, { useState } from 'react';

export default function TAForm({ handleSubmit }) {
  const [ name, setName ] = useState('')
  const [ cohort, setCohort ] = useState('')
    return (
      <>
        <h2>Add New TA</h2>
        <form id='taForm' onSubmit={handleSubmit.bind(this, name, cohort)}>
          <label>TA Name</label>
          <input type='text' name='name' value={name} onChange={({target}) => setName(target.value)}/>
  
          <label>Cohort</label>
          <input type='text' name='cohort' value={cohort} onChange={({target}) => setCohort(target.value)}/>

          <button type='submit'>Submit</button>
        </form>
      </>
    );
}