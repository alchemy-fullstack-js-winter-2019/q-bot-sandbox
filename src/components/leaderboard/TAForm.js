import React, { useState } from 'react';
import CohortSort from '../CohortSort';

export default function TAForm({ user, handleSubmit }) {
  const [ name ] = useState('')
  const [ cohort, setCohort ] = useState('')
    return (
      <>
        <h2 className={'addTaHeader'}>Add New TA</h2>
        <form id='taForm' className={'taForm'} onSubmit={handleSubmit.bind(this, name, cohort)}>
          <label>TA Name</label>
          <input 
            type='text' 
            name='name' 
            value={user || 'User'} 
            disabled="disabled"
          />

          <CohortSort onChange={({target}) => setCohort(target.value)} />

          <button className={'submit'} type="submit">Submit</button>
        </form>
        
      </>
    );
}
