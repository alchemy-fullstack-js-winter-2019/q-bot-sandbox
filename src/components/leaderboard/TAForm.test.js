import React from 'react';
import { shallow } from 'enzyme';
import TAForm from './TAForm';

describe('TAForm', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <TAForm user={'User Name'}
        handleSubmit={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
