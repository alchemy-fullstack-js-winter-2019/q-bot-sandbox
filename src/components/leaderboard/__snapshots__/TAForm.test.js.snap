// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TAForm matches a snapshot 1`] = `
<Fragment>
  <h2
    className="addTaHeader"
  >
    Add New TA
  </h2>
  <form
    className="taForm"
    id="taForm"
    onSubmit={[Function]}
  >
    <label>
      TA Name
    </label>
    <input
      disabled="disabled"
      name="name"
      type="text"
      value="User Name"
    />
    <label>
      Cohort
    </label>
    <CohortSort
      onChange={[Function]}
    />
    <button
      className="submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</Fragment>
`;
